ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -201,9 +203,14 @@ data InternalState m peer blk = IS {
     , varCandidates       :: TVar m (Map peer (TVar m (CandidateState blk)))
     , varState            :: TVar m (NodeState (BlockProtocol blk))
     , tracer              :: Tracer m String
+    , forgeTracer         :: Tracer m (TraceForgeEvent blk)
     , mempool             :: Mempool m blk TicketNo
     }
 
+data TraceForgeEvent blk =
+    -- | The forged block.
+    TraceForgeEvent blk
```

Comment:
Can you include the `SlotNo` as well? Then this could replace the existing `String`-based trace message.

Eventually, we'll want to clean up the tracing in the node (no more `String`s).

Reply:
Done